packages/data-stores/src/queries/use-has-seen-whats-new-modal-query.ts

```diff
@@ -20,11 +20,7 @@ export const useHasSeenWhatsNewModalQuery = ( siteId: number | null ) => {
 
 	const { data, isLoading } = useQuery< { has_seen_whats_new_modal: boolean } >(
 		queryKey,
-		() =>
-			wpcomRequest( {
-				path: `/sites/${ siteId }/block-editor/has-seen-whats-new-modal`,
-				apiNamespace: 'wpcom/v2',
-			} ),
+		() => apiFetch( { path: `/wpcom/v2/block-editor/has-seen-whats-new-modal` } ),
```

Comment:
I see that the `queryKey` (which is used for caching the response) doesn't take the site ID into account, which means I don't think the hook would have actually prefetched when the site ID changed. With ReactQuery if you want it to fetch again you'd need something like `queryKey = [ 'has-seen-whats-new-modal', siteId ]`. At the moment if the site ID changes ReactQuery will see that it already has something in it's cache for this value and not perform a refetch.

> I see where siteId is being used to enable refetch on the hook

If we did ever want to support a refetch then ReactQuery has a feature for this. [`useQuery()`returns a `refetch()` function](https://react-query.tanstack.com/reference/useQuery) which can be called to explicitly refetch. But I can't see it ever being needed for this hook.

> siteId was not only used in the url but in hook configuration also

It's used to disable the query if the ID isn't available in the component yet, which can probably happen during component initialisation. But I think that's only to prevent invalid URLs being used. Given the hook doesn't depend on the ID to generate the URL anymore I don't think the query should be disabled based on something it's not going to use anyway.